addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1478,6 +1509,193 @@ export class ListRenderer extends Component {
+            fields.push({
+                ...baseFieldObject,
+                FieldComponent: PropertyValue,
+                rawAttrs: {
+                    ...baseFieldObject.rawAttrs,
+                    name: propertyName,
+                },
+                name: propertyName,
+                label: property.string,
+                propertyType: property.type,
+                string: property.string,
+                propertyParentId: property.parentId, //used to group related columns
+                propertyParentName: property.parentName,
+            });
+        }
+        return fields;
+    }
+
+    /**
+     * Get the columns defined for the records with additional virtual columns
+     * Create for the different property values of the 'properties' field
+     *
+     * @param {Array} columns all the columns (typically props.archInfo.columns)
+     * @param {Array} propertyDefinitions relevant data to generate property 'fields'
+     * @returns {Array} the same array with the 'properties' columns replaced by 1 column per property
+     */
+    propertiesGetAllColumns(columns, propertyDefinitions) {
+        columns = [...columns];
+        const propertiesColumnIndices = [];
+        const propertiesFieldNames = Object.keys(propertyDefinitions);
+        for (let i = 0; i < columns.length; i++) {
+            const column = columns[i];
+            if (propertiesFieldNames.includes(column.name)) {
+                propertiesColumnIndices.push(i);
+            }
+        }
+        //indices are assumed sorted
+        let rightOffset = 0;
+        for (const columnIndex of propertiesColumnIndices) {
+            const correctedColumnIndex = columnIndex+rightOffset
+            const baseColumnObject = columns[correctedColumnIndex];
+            const propertyColumns = this.getVirtualPropertyFields(baseColumnObject, propertyDefinitions[baseColumnObject.name]);
+            for (let i = 0; i < propertyColumns.length; i++) {
+                propertyColumns[i].id = `${columnIndex}_property_${i}`;
+            }
+            columns.splice(correctedColumnIndex, 1, ...propertyColumns);
+            rightOffset = rightOffset - 1 + propertyColumns.length;
+        }
+        return columns;
+    }
+
+    /**
+     * Replace 'properties' field with a field for each property value
+     * preserving most of the attributes of the original field.
+     *
+     * @param {object} fields in the form of fieldInfo (e.g. this.props.list.fields)
+     * @param {array} propertyDefinitions list of relevant data about the property 'fields'
+     * @returns {object} input or altered copy of the input
+     */
+    propertiesProcessFields(fields, propertyDefinitions) {
+        const propertiesFieldNames = Object.keys(propertyDefinitions);
+        if (!propertiesFieldNames) {
+            return fields;
+        }
+        fields = {...fields};
+        for (const propertiesFieldName of propertiesFieldNames) {
```

Comment:
do you think it's possible to use the same `flatMap` trick here ?

Reply:
I don't think it's useful here since I work with a hashmap and a list of 2/3 elements tops.

Tried turning it into a map but since I need 3 lines just to create an object with a dynamic name I don't think it gets much clearer.

```
const propertiesFields = propertiesFieldNames.map(name => fields[name]).filter(field => field);
        for (const propertiesField of propertiesFields) {
                {fields,
                ...this.getVirtualPropertyFields(propertiesField, propertyDefinitions[propertiesFieldName])
                .map(property => {
                    const ret = {}
                    ret[property.name] = property;
                    return ret;
                })
...
```